x-pack/plugins/observability_solution/infra/public/components/asset_details/tabs/dashboards/dashboards.tsx

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { i18n } from '@kbn/i18n';
+import {
+  EuiPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiSpacer,
+  EuiEmptyPrompt,
+  EuiLoadingLogo,
+} from '@elastic/eui';
+
+import { ViewMode } from '@kbn/embeddable-plugin/public';
+import {
+  AwaitingDashboardAPI,
+  DashboardCreationOptions,
+  DashboardRenderer,
+} from '@kbn/dashboard-plugin/public';
+
+import { buildPhraseFilter, Filter } from '@kbn/es-query';
+import type { DataView } from '@kbn/data-views-plugin/common';
+import { HOST_FIELD } from '../../../../../common/constants';
+import type {
+  DashboardIdItem,
+  DashboardItemWithTitle,
+  InfraCustomDashboardAssetType,
+} from '../../../../../common/custom_dashboards';
+
+import { EmptyDashboards } from './empty_dashboards';
+import { EditDashboard, GotoDashboard, LinkDashboard, UnlinkDashboard } from './actions';
+import { useCustomDashboard } from '../../hooks/use_custom_dashboards';
+import { useDatePickerContext } from '../../hooks/use_date_picker';
+import { useAssetDetailsRenderPropsContext } from '../../hooks/use_asset_details_render_props';
+import { FETCH_STATUS, useDashboardFetcher } from '../../hooks/use_dashboards_fetcher';
+import { useDataViewsContext } from '../../hooks/use_data_views';
+import { DashboardSelector } from './dashboard_selector';
+import { ContextMenu } from './context_menu';
+import { useAssetDetailsUrlState } from '../../hooks/use_asset_details_url_state';
+
+const fieldByAssetType = {
+  host: HOST_FIELD,
+} as Record<InfraCustomDashboardAssetType, string>;
+
+export function getFilterByAssetName(
```

Comment:
Can we unit test this?

Reply:
It executes the `buildPhraseFilter` if there are valid arguments (`assetNameField` coming from a `dataView` method) and pushes an item to an array, so the main logic is in `buildPhraseFilter` - what do you think should be tested here?

Reply:
Well, you can still validate when a valid and an invalid arguments are passed.

Reply:
I added tests to check when we have the field name returned from the data view and when it's undefined (so for both returning a filter and returning empty array cases)